### PR TITLE
Add react-native-onesignal

### PR DESCRIPTION
**Pull request by FabianMeul**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by FabianMeul** [comment]
Big thanks to @krystofcelba for providing the initial types in https://github.com/geektimecoil/react-native-onesignal/pull/395

File: types/react-native-onesignal/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+export interface OneSignal {
+    /**
+     * Initialize Onesignal
+     */
+    init(appId: string, settings?: Settings): void;
+
+    /**
+     * Undocumented function.
+     */
+    Configure(undocumentedParam1?: any, undocumentedParam2?: any): void;
```
Comment (slorber) [inline types/react-native-onesignal/index.d.ts]: this is normally lowercase no?

File: types/react-native-onesignal/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    Warnings = 3,
+    Info = 4,
+    Debug = 5,
+    Verbose = 6,
+}
+
+export type oneSignalEvents = 'received' | 'opened' | 'ids';
+
+export interface Settings {
+    kOSSettingsKeyAutoPrompt: boolean;
```
Comment (slorber) [inline types/react-native-onesignal/index.d.ts]: make all attributes optional with ? please

**Review by slorber — CHANGES_REQUESTED** [review]
These types seem to work great for me except few details

Note there is a method `syncHashedEmails` that is missing

File: types/react-native-onesignal/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+    /**
+     * OneSignal now allows you to send emails to your userbase. This email can be set using the OneSignal react-native SDK.
+     */
+    setEmail(email: string, emailAuthCode: string, callback: (error?: any) => {}): void;
+
+    /**
+     * If you don't want to implement email auth hashing on your backend (which is heavily recommended), you can still use the OneSignal email feature in an unauthenticated state with this function.
+     */
+    setEmail(email: string, callback: (error?: any) => {}): void;
```
Comment (slorber) [inline types/react-native-onesignal/index.d.ts]: I think the callback should be optional

File: types/react-native-onesignal/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+    /**
+     * You can delete tag from user with this function.
+     */
+    deleteTag(key: string): void;
+
+    /**
+     * OneSignal now allows you to send emails to your userbase. This email can be set using the OneSignal react-native SDK.
+     */
+    setEmail(email: string, emailAuthCode: string, callback: (error?: any) => {}): void;
```
Comment (slorber) [inline types/react-native-onesignal/index.d.ts]: optional callback?

**Pull request closed without merging**